007-SIP2-hold_pickup.patch

This patch modifies the SIP return message on item information request for holds.
If there is a 'hold_shelf' property on the item, we add cardnumber and pickupnumber as
item properties (CH) in the response, so it can be parsed by client or proxy

diff --git a/C4/SIP/Sip/MsgType.pm b/C4/SIP/Sip/MsgType.pm
index ae802b9..5ae4fc0 100644
--- a/C4/SIP/Sip/MsgType.pm
+++ b/C4/SIP/Sip/MsgType.pm
@@ -1174,6 +1174,27 @@ sub handle_item_information {
             $resp .= add_field( FID_HOLD_PICKUP_DATE, timestamp($i) );
         }
 
+        # Deichman mod: return patron and pickupnumber in CH (item properties) if item is marked hold_shelf
+        # Need to be parsed by sip_proxy to check response against "active" patron, as Self checkout machine
+        # has no concept of patron session.
+        # if they match - hold is waiting for this patron
+        # if not - hold is for another patron, so ignore it
+        if ( scalar @{ $item->hold_shelf } > 0 ) {
+            use Koha::Holds;
+            use Koha::Patrons;
+            warn Dumper($item);
+            # We must assume hold_shelf only has one item waiting (found='W')
+            my $waiting = $item->hold_shelf->[0];
+            my $patron = Koha::Patrons->find($waiting->{'borrowernumber'});
+            if ($patron) {
+                $resp .= add_field( FID_ITEM_PROPS, "CardNumber:" . $patron->cardnumber );
+            }
+            my $hold = Koha::Holds->find($waiting->{'reserve_id'});
+            if ($hold) {
+                $resp .= add_field( FID_ITEM_PROPS, "PickupNumber:" . $hold->pickupnumber );
+            }
+        }
+
         $resp .= maybe_add( FID_SCREEN_MSG, $item->screen_msg, $server );
         $resp .= maybe_add( FID_PRINT_LINE, $item->print_line );
     }